src/utils.js

```diff
@@ -39,26 +39,37 @@ class HtmlElementTypeError extends Error {
   }
 }
 
-function checkHasWindow(htmlElement, ...args) {
+function getWindow(htmlElement, ...args) {
   if (
-    !htmlElement ||
-    !htmlElement.ownerDocument ||
-    !htmlElement.ownerDocument.defaultView
+    htmlElement &&
+    htmlElement.ownerDocument &&
+    htmlElement.ownerDocument.defaultView
   ) {
-    throw new HtmlElementTypeError(htmlElement, ...args)
+    return htmlElement.ownerDocument.defaultView
+  }
+
+  if (htmlElement && htmlElement.defaultView) {
+    return htmlElement.defaultView
   }
+
+  throw new HtmlElementTypeError(htmlElement, ...args)
 }
 
-function checkHtmlElement(htmlElement, ...args) {
-  checkHasWindow(htmlElement, ...args)
-  const window = htmlElement.ownerDocument.defaultView
+function getHtmlElement(htmlElement, ...args) {
```

Comment:
Why do we need to re-purpose this? This is now used to reassigning function arguments which is always a bit iffy. It's just the identity function but if it isn't necessary then why introduce it in the first place?